src/main/java/se/kth/hopsworks/message/Message.java

```diff
@@ -0,0 +1,281 @@
+import javax.persistence.Lob;
+import javax.persistence.ManyToMany;
+import javax.persistence.ManyToOne;
+import javax.persistence.NamedQueries;
+import javax.persistence.NamedQuery;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+import javax.persistence.Temporal;
+import javax.persistence.TemporalType;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import javax.xml.bind.annotation.XmlRootElement;
+import javax.xml.bind.annotation.XmlTransient;
+import org.codehaus.jackson.annotate.JsonIgnore;
+import se.kth.hopsworks.user.model.Users;
+
+@Entity
+@Table(name = "hopsworks.message")
+@XmlRootElement
+@NamedQueries({
+  @NamedQuery(name = "Message.findAll",
+          query
+          = "SELECT m FROM Message m"),
+  @NamedQuery(name = "Message.findById",
+          query
+          = "SELECT m FROM Message m WHERE m.id = :id"),
+  @NamedQuery(name = "Message.findByDateSent",
+          query
+          = "SELECT m FROM Message m WHERE m.dateSent = :dateSent"),
+  @NamedQuery(name = "Message.findByFrom",
+          query
+          = "SELECT m FROM Message m WHERE m.from = :from"),
+  @NamedQuery(name = "Message.findMessagesByToAndUnread",
+          query
+          = "SELECT m FROM Message m WHERE m.unread = :unread AND m.to = :to"),
+  @NamedQuery(name = "Message.countByToAndUnread",
+          query
+          = "SELECT COUNT(m.id) FROM Message m WHERE m.to = :to AND m.unread = :unread"),
+  @NamedQuery(name = "Message.findByToAndDeleted",
+          query
+          = "SELECT m FROM Message m WHERE m.to = :to AND m.deleted = :deleted  ORDER BY m.unread DESC, m.dateSent DESC"),
+  @NamedQuery(name = "Message.emptyToAndDeleted",
+          query
+          = "DELETE FROM Message m WHERE m.to = :to AND m.deleted = :deleted"),
+  @NamedQuery(name = "Message.countByToAndDeleted",
+          query
+          = "SELECT COUNT(m.id) FROM Message m WHERE m.to = :to AND m.deleted = :deleted")})
+public class Message implements Serializable {
+
+  private static final long serialVersionUID = 1L;
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  @Basic(optional = false)
+  @Column(name = "id")
+  private Integer id;
+  @Basic(optional = false)
+  @NotNull
+  @Column(name = "date_sent")
+  @Temporal(TemporalType.TIMESTAMP)
+  private Date dateSent;
+  @Size(max = 128)
+  @Column(name = "subject")
+  private String subject;
+  @Size(max = 128)
+  @Column(name = "preview")
+  private String preview;
+  @Basic(optional = false)
+  @NotNull
+  @Lob
+  @Size(min = 1,
+          max = 65535)
+  @Column(name = "content")
+  private String content;
+  @Basic(optional = false)
+  @NotNull
+  @Column(name = "unread")
+  private boolean unread;
+  @Basic(optional = false)
+  @NotNull
+  @Column(name = "deleted")
+  private boolean deleted;
+  @Size(max = 600)
+  @Column(name = "path")
+  private String path;
+  @JoinTable(name = "hopsworks.message_to_user",
```

Comment:
This could be a problem for us when the number of messages per user grows. NDB is *not* good at joins. They are expensive and become slower the larger the cluster is....
We can leave it in, but keep it in mind, and maybe do some measurements with 1000s of messages for a user and time it.